code/modules/projectiles/gun.dm

```diff
@@ -173,6 +176,29 @@
 
 	return 1
 
+/obj/item/weapon/gun/verb/wield_gun()
+	set name = "Wield Firearm"
+	set category = "Object"
+	set src in usr
+
+	if(is_wieldable)
+		toggle_wield(usr)
+		update_held_icon()
+	else
+		to_chat(usr, "<span class='warning'>You can't wield \the [src]!</span>")
+
+/obj/item/weapon/gun/ui_action_click()
+	if(src in usr)
+		wield_gun()
+
+/obj/item/weapon/gun/proc/update_wield_verb()
+	if(is_wieldable) //If the gun is marked as wieldable, make the action button appear and add the verb.
+		action_button_name = "Wield Firearm"
+		verbs += /obj/item/weapon/gun/verb/wield_gun
+	else
+		verbs -= /obj/item/weapon/gun/verb/wield_gun

```

Comment:
```suggestion
		action_button_name = ""
		verbs -= /obj/item/weapon/gun/verb/wield_gun
```

Gotta assume this gets called in a fashion like TRUE -> FALSE, at which case, we gotta null this var as well.